cg/apps/tb/models.py

```diff
@@ -1,49 +1,41 @@
-import datetime as dt
-from pathlib import Path
 from typing import Optional
 
-from dateutil.parser import parse as parse_datestr
-from pydantic.v1 import BaseModel, validator
+from pydantic import AfterValidator, BaseModel, ConfigDict, FieldValidationInfo, field_validator
+from typing_extensions import Annotated
+
+from cg.apps.tb.validators import parse_str_to_datetime, parse_str_to_path
 
 
 class TrailblazerAnalysis(BaseModel):
     id: int
     family: str
-    case_id = Optional[str]
-
-    version: Optional[str]
-    logged_at: Optional[str]
-    started_at: Optional[str]
-    completed_at: Optional[str]
-    status: Optional[str]
-    priority: Optional[str]
-    out_dir: Optional[str]
-    config_path: Optional[str]
-    comment: Optional[str]
-    is_deleted: Optional[bool]
-    is_visible: Optional[bool]
-    type: Optional[str]
-    user_id: Optional[int]
+    case_id: Optional[str] = None
+
+    @field_validator("case_id")
+    @classmethod
+    def inherit_family_value(cls, value: str, info: FieldValidationInfo) -> str:
+        return info.data.get("family")
+
+    version: Optional[str] = None
+    logged_at: Annotated[Optional[str], AfterValidator(parse_str_to_datetime)]
+    started_at: Annotated[Optional[str], AfterValidator(parse_str_to_datetime)]
+    completed_at: Annotated[Optional[str], AfterValidator(parse_str_to_datetime)]
+    status: Optional[str] = None
+    priority: Optional[str] = None
+    out_dir: Annotated[Optional[str], AfterValidator(parse_str_to_path)]
+    config_path: Annotated[Optional[str], AfterValidator(parse_str_to_path)]
```

Comment:
Same thing with out_dir and config_path where I think it would make more sense to have a BeforeValidator converting the values to Paths and type hinting them as such.

Reply:
Nice 👍